pkg/services/authn/clients/jwt.go

```diff
@@ -75,13 +76,24 @@ func (s *JWT) Authenticate(ctx context.Context, r *authn.Request) (*authn.Identi
 			SyncTeams:       s.cfg.JWTAuth.GroupsAttributePath != "",
 		}}
 
+	serializedClaims, _ := json.Marshal(&claims)
```

Comment:
Instead of marshaling/unmarshaling you can use `map[string]any(claims)` in the code. 
Example:
```
id.Email, _ = util.SearchJSONForStringAttr(s.cfg.JWTAuth.EmailAttributePath, map[string]any(claims))
```
Because `claims`'s type is `JWTClaims` which is a "redefinition of `DynMap` -> `map[string]any`, so because of this the underlying jmespath lib cannot work without the type conversion to `map[string]any`, because it checks for `map[string]any`, but it receives `JWTClaims`.

I don't think the custom type definitions (without any functions attached to those) provides any benefit here. So we could remove `JWTClaims` and use `map[string]any` instead to make the code simpler. @kalleep wdyt?

Reply:
Yeah I think it makes sense, either solution would work but I think I would prefer to remove the custom definition `JwtClaims` and just return map[string]any for the reasons @mgyongyosi stated

Reply:
As suggested, I have replaced `JWTClaims` with `map[string]any` and removed the conversation of claims to map. Looks much cleaner!